net/ipv4/tcp_output.c

```diff
@@ -2648,6 +2648,8 @@ tcp_tfw_sk_prepare_xmit(struct sock *sk, struct sk_buff *skb,
 		*limit -= TLS_MAX_OVERHEAD;
 
 	if (unlikely(skb_tfw_flags(skb) & SS_F_HTTP2_FRAME_PREPARED)) {
+		if (skb->len > *limit)
+			return -ENOMEM;
```

Comment:
You return `-ENOMEM` that leads to reusing current skb, can't be a problem here when skb with frames was prepared with large limit then error occurred and after max limit was changed(for instance due to changing mtu)? Looks like in such case we will retry to send current skb in a loop very long.